src/particles/elements/Drift.H

```diff
@@ -92,6 +107,8 @@ namespace impactx
             py = pyout;
             pt = ptout;
 
+            // undo shift due to alignment errors of the element
+            shift_out(x, y, px, py);
         }
```

Comment:
Found a bug--after the application of shift_out, the values of x and y need to assigned to p.pos(RealAoS::x) and p.pos(RealAoS::y), respectively.  See the suggestions above.  There may be a cleaner way to re-work this, and the pattern will need to be repeated for all elements.

Reply:
Oh, you are right!
x and y are *not* yet passed into the push by reference :-o

Reply:
Fix pushed :)